eclass/gap-pkg.eclass

```diff
@@ -0,0 +1,388 @@
+	# involving dev-gap/browse is likely to bork the user's terminal.
+	# The "browse" package is however smart enough to figure out when
+	# stdout is not a tty, and avoids breaking it in that case. So by
+	# piping to tee, we encourage it not to do anything too crazy.
+	${gapcmd} | tee test-suite.log \
+		|| die "test suite failed, see test-suite.log"
+}
+
+# @ECLASS_VARIABLE GAP_PKG_EXTRA_INSTALL
+# @DEFAULT_UNSET
+# @DESCRIPTION
+# A bash array of extra files and directories to install recursively at
+# the root of this package's directory tree. For example, if you have a
+# package that mostly follows the suggested layout (described in the
+# gap-pkg_src_install documentation) but also includes a "data"
+# directory, you should set
+#
+#   GAP_PKG_EXTRA_INSTALL=( data )
+#
+# to install the data directory without having to override the entire
+# src_install phase.
+
+# @ECLASS_VARIABLE GAP_PKG_HTML_DOCDIR
+# @DEFAULT_UNSET
+# @DESCRIPTION
+# The directory inside the tarball where the HTML documentation is
+# located. This is _usually_ "doc", which conforms to the suggested
+# GAPDoc layout and is where we look if this variable is left
+# unset. Many packages however use a top-level "htm" directory
+# instead. The named directory will be installed to gap-pkg_dir and
+# symlinked to the usual location under /usr/share/doc. As a result, you
+# should only use this for directories referenced by PackageInfo.g or by
+# some other part of the package.  HTML documentation whose location
+# doesn't need to be known to the package at runtime should instead be
+# installed with HTML_DOCS or a similar mechanism.
+
+# @FUNCTION: gap-pkg_src_install
+# @USAGE:
+# @DESCRIPTION:
+# Install a GAP package that follows the suggested layout,
+#
+#   https://docs.gap-system.org/doc/ref/chap76.html
+#
+# In particular:
+#
+# 1. All GAP source files (*.g) in $S are installed.
+#
+# 2. If a library directory named "gap" or "lib" exists,
+#    it is installed.
+#
+# 3. If a binary directory "bin" exists, it is installed.
+#
+# 4. If a "doc" directory exists, we assume GAPDoc conventions
+#    (https://docs.gap-system.org/pkg/gapdoc/doc/chap5.html) and install
+#    what we find there. Unfortunately for us, each package's
+#    PackageInfo.g contains a "PackageDoc" section that points to this
+#    documentation, and we can't break the paths it references. Instead,
+#    we try to dosym the human-readable parts of the documentation (PDF
+#    manuals) into appropriate Gentoo locations.
+#
+# 5. We consult GAP_PKG_HTML_DOCDIR for the HTML documentation and repeat
+#    the process above.
+#
+# A few GAP packages have autotools build systems with working "make
+# install" routines, but most don't. So for the time being we omit that
+# step. It's harder to work around the packages that don't support it
+# than the other way around.
+gap-pkg_src_install() {
+	einstalldocs
+
+	# Install the "normal" documentation: pdf, six, and txt files under doc/
```

Comment:
The file type list here does not match the actual list in the loop?

Reply:
Yeah, I don't know why I tried to repeat an implementation detail in a comment. I know better. I've changed it.